libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1230,6 +1246,28 @@ string YulUtilFunctions::resizeDynamicByteArrayFunction(ArrayType const& _type)
 	});
 }
 
+string YulUtilFunctions::cleanUpDynamicByteArrayEndSlotsFunction(ArrayType const& _type)
+{
+	solAssert(_type.isByteArray(), "");
+	solAssert(_type.isDynamicallySized(), "");
+
+	string functionName = "clean_up_byte_array_end_slots_" + _type.identifier();
+	return m_functionCollector.createFunction(functionName, [&](vector<string>& _args, vector<string>&) {
+		_args = {"slot", "len", "pos"};
+		return Whiskers(R"(
+			if gt(len, 31) {
+				let dataArea := <dataLocation>(slot)
+				let deleteStart := add(dataArea, div(add(pos, 31), 32))
```

Comment:
Nit: didn't we had a `roundUp32()` function? We may want to decide to use it everywhere or not use it at all.

It probably should not be done in this PR though as it is clearer now to see these statements have only been moved.

Reply:
We have roundUp function, and I updated this piece of code to use it.